source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/credentials/CredentialsManagerFactory.java

```diff
@@ -90,6 +91,20 @@ public static CredentialsManager getCredentialsManager(
             }
         }
 
+        if (Platform.isCurrentPlatform(Platform.LINUX)) {
+            /*
+             * Linux uses gnome-keyring if it's available unless the user
+             * specifies using PersistanceStoreCredentialsManager
+             */
+            if (usePersistanceCredentialsManager) {
+                return new PersistenceStoreCredentialsManager(persistenceProvider.getConfigurationPersistenceStore());
+            } else {
```

Comment:
I would refactor a bit the code of this method. Too many branches return one and the same new PersistenceStoreCredentialsManager(persistenceProvider.getConfigurationPersistenceStore())